src/openforms/authentication/contrib/digid/tests/test_csp_update.py

```diff
@@ -0,0 +1,125 @@
+from pathlib import Path
+from unittest.mock import patch
+
+from django.test import TestCase, override_settings
+from django.urls import reverse
+
+from digid_eherkenning.models.digid import DigidConfiguration
+from privates.test import temp_private_root
+from simple_certmanager.test.factories import CertificateFactory
+
+from openforms.config.constants import CSPDirective
+from openforms.config.models import CSPSetting
+from openforms.forms.tests.factories import (
+    FormDefinitionFactory,
+    FormFactory,
+    FormStepFactory,
+)
+from openforms.utils.tests.cache import clear_caches
+
+TEST_FILES = Path(__file__).parent / "data"
+METADATA_POST = TEST_FILES / "metadata_POST_bindings.xml"
+
+
+@temp_private_root()
+@override_settings(CORS_ALLOW_ALL_ORIGINS=True, IS_HTTPS=True)
+class CSPUpdateTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        cert = CertificateFactory.create(label="DigiD", with_private_key=True)
+
+        cls.config = DigidConfiguration.get_solo()
+
+        cls.config.certificate = cert
+        cls.config.idp_service_entity_id = "https://test-digid.nl"
+        cls.config.want_assertions_signed = False
+        cls.config.entity_id = "https://test-sp.nl"
+        cls.config.base_url = "https://test-sp.nl"
+        cls.config.service_name = "Test"
+        cls.config.service_description = "Test description"
+        cls.config.slo = False
+        cls.config.save()
+
+    def setUp(self):
+        super().setUp()
+
+        clear_caches()
+        self.addCleanup(clear_caches)
+
+    @patch(
+        "onelogin.saml2.idp_metadata_parser.OneLogin_Saml2_IdPMetadataParser.get_metadata"
+    )
+    def test_csp_updates(self, get_metadata):
+        # assert no csp entries exist with initial solo model
+        self.assertTrue(CSPSetting.objects.none)
+
+        # assert new csp entry is added after adding metadata url
+        with METADATA_POST.open("rb") as md_file:
+            metadata_content = md_file.read().decode("utf-8")
```

Comment:
pathlib has some nicer methods for this:

```py
metadata_content = METADATA_POST.read_text()
```
if I remember correctly. It uses a context manager internally, so less indentation to handle in your own code :)